b2share/modules/schemas/cli.py

```diff
@@ -46,3 +64,225 @@ def init(verbose):
+    try:
+        block_schema = BlockSchema.get_block_schema(schema_id=block_schema_id)
+    except BlockSchemaDoesNotExistError:
+        raise click.BadParameter("No block_schema with id %s" % block_schema_id)
+    click.secho("BLOCK SCHEMA VERSIONS FOR community %s, block schema %s" % (Community.get(id=block_schema.community).name, block_schema.name))
+    click.secho("Version no.\tRelease date")
+    for bl_schema_version in block_schema.versions:
+        click.secho("%s\t%s" % (bl_schema_version.version, bl_schema_version.released))
+
+@schemas.command()
+@with_appcontext
+@click.option('-v','--verbose', is_flag=True, default=False)
+@click.argument('block_schema_id')
+@click.option('--version')
+def block_schema_version_generate_json(verbose, block_schema_id, version=None):
+    """print json_schema of a particular block schema version."""
+    try:
+        val = UUID(block_schema_id, version=4)
+    except ValueError:
+        raise click.BadParameter("BLOCK_SCHEMA_ID is not a valid UUID (hexadecimal numbers and dashes e.g. fa52bec3-a847-4602-8af5-b8d41a5215bc )")
+    try:
+        block_schema = BlockSchema.get_block_schema(schema_id=block_schema_id)
+    except BlockSchemaDoesNotExistError:
+        raise click.BadParameter("No block_schema with id %s" % block_schema_id)
+    if version:
+        click.secho( block_schema.versions[version].json_schema )
+    else:
+        click.secho(block_schema.versions[len(block_schema.versions)-1].json_schema)
+
+@schemas.command()
+@with_appcontext
+@click.option('-v','--verbose', is_flag=True, default=False)
+@click.argument('community')
+@click.option('--version')
+def community_schema_list_block_schema_versions(verbose, community, version=None):
+    """Show the block schema versions in the community schema element of a specific community"""
+    comm = get_community_by_name_or_id(community)
+    if not comm:
+        raise click.BadParameter("There is no community by this name or ID: %s" % community)
+    community_schema = CommunitySchema.get_community_schema(comm.id, version)
+    if not community_schema:
+        raise click.ClickException("Community %s does not have a community schema" % comm.name)
+    community_schema_dict = json.loads(community_schema.community_schema)
+    props = community_schema_dict['properties']
+    click.secho("The following block schema versions are listed for community %s, community schema version %s" % (comm.name, "latest %d" % community_schema.version if not version else version))
+    for key in props:
+        click.secho("Block schema: %s, version url: %s" % (key, props[key]['$ref']))
+    
+@schemas.command()
+@with_appcontext
+@click.option('-v','--verbose', is_flag=True, default=False)
+@click.argument('community')
+@click.argument('block_schema_id')
+@click.option('--version')    
+def community_schema_add_block_schema_version(verbose, community, block_schema_id, version=None):
```

Comment:
I am not sure about `community_schema_remove_block_schema_version` and `community_schema_add_block_schema_version`. It adds features which
1/ are not supported by the programmatic and REST APIs.
2/ we don't know if the admin will have a need for. The CLI is **only** for the admin/superuser having direct access to the b2share server.

Also the CLI does not have the simpler community schema edition functions provided by the programmatic and REST APIs, i.e. just giving json directly with the root schema version number. Thus it is not possible to later use "AllOf", "AnyOf", "OneOf" or more complex structures. It is also not possible to update the root schema version used by a community schema.

I understand the need for simplicity, and I agree that there is a use case for "adding" and "removing" block schemas to community schemas. I am just not sure that this is the right moment and place to add this feature. There is a risk of constraining what we can do in the future by implementing it now like that.

A simpler step, in my humble and subjective opinion, would be to do the first implementation of this advanced feature in the UI. We can suppose that the admin knows a little about JSON and will be able to write the JSON Schema if he needs to.  That way the REST API and the CLI remain simple and we can migrate the feature in the backend later once we have more user feedback, and if it is needed.